test/unit/tasks/migrate_alt_text_to_caption_test.rb

```diff
@@ -0,0 +1,87 @@
+require "test_helper"
+
+class MigrateAltTextToCaptionTest < ActiveSupport::TestCase
+  setup do
+    @task = Rake::Task["data_migration:alt_text_to_caption"]
+    @task.reenable
+  end
+
+  test "migration task is idempotent" do
+    image1 = FactoryBot.create(:image, alt_text: "Test alt text", caption: "")
+    image2 = FactoryBot.create(:image, alt_text: "Another alt", caption: "Existing caption")
+    image3 = FactoryBot.create(:image, alt_text: nil, caption: "No alt text")
+
+    @task.invoke
+
+    image1.reload
+    image2.reload
+    image3.reload
+
+    assert_nil image1.alt_text, "Alt text should be cleared after migration"
+    assert_equal "Test alt text", image1.caption
+
+    assert_nil image2.alt_text, "Alt text should be cleared after migration"
+
+    assert_nil image3.alt_text, "Alt text should remain nil"
+    assert_equal "No alt text", image3.caption, "Caption should be unchanged"
+
+    first_run_captions = [image1.caption, image2.caption, image3.caption]
+
+    @task.reenable
+    @task.invoke
+
+    image1.reload
+    image2.reload
+    image3.reload
+
+    assert_equal first_run_captions, [image1.caption, image2.caption, image3.caption],
+                 "Second run should not change any captions"
+
+    assert_nil image1.alt_text, "Alt text should still be nil"
+    assert_nil image2.alt_text, "Alt text should still be nil"
+    assert_nil image3.alt_text, "Alt text should still be nil"
+  end
+
+  test "migration task only processes images with alt text" do
```

Comment:
Similarly, it would be good to 'listen' to the factory objects here and assert that `update`/`update_column`/whatever isn't even called on the images we're expecting to be skipped over.